rfcs/02-coresets.md

```diff
@@ -0,0 +1,47 @@
+# Coreset API
+
+
+Coresets (or core-sets) are methods that tries to select of subset of a dataset that will best represent the true distribution.
+
+## Proposition
+
+Coresets are similar to heuristics, but instead of selecting data based on uncertainty, they would use features.
+
+Gathering features is easy, we can use a `forward_hook`.
+
+**API**
+```python
+Array = np.ndarray
+Indices = List[int]
+
+class BaseCoreset:
+    def get_ranks(features: Array, logits: Array) -> Indices:
+        # K-Means or something
+        pass
+        
+    def filter(logits, heuristic, proportion=1.)-> Indices:
+        # Use `heuristic` to rank features and return the top `proportion` indices.
+        pass
```

Comment:
should we have a way to give more importance to coreset results or to heuristic results?

Reply:
Not sure how to merge them.
The idea of this method is to use BALD to create a pool of potential candidates (ie. the top 10%) and select from them with a coreset. I will remove it as we don't know if it works.